### PR TITLE
Adds check for existing /current directory to deploy:prepare

### PR DESCRIPTION
**Pull request by vicdelfant**
| Q             | A
| ------------- | ---
| Bug fix?      | Yes
| New feature?  | No
| BC breaks?    | No
| Deprecations? | No
| Fixed tickets | N/A

Re-submission of #583, but instead of enforcing the removal of this directory we'll now throw an exception.

### CODE REVIEW

File: recipe/common.php [hunk 1]
```diff
@@ -103,6 +103,12 @@
 
     run('if [ ! -d {{deploy_path}} ]; then mkdir -p {{deploy_path}}; fi');
 
+    // Check for existing /current directory (not symlink)
+    $result = run('if [ ! -L {{deploy_path}}/current ] && [ -d {{deploy_path}}/current ]; then echo 1; fi')->toString();
```
Comment (elfet) [inline recipe/common.php]: It's better to use `toBool()`:
```php
run('if [...]; then echo true; fi')->toBool();
```

File: recipe/common.php [hunk 2]
```diff
@@ -103,6 +103,12 @@
...
+    if ($result) {
+        throw new RuntimeException('There already is a directory (not symlink) named "current" in the deploy_path. Remove this directory so it can be replaced with a symlink for atomic deployments.');
```
Comment (elfet) [inline recipe/common.php]: Please, insert real deploy path in error message.

File: recipe/common.php [hunk 3]
```diff
@@ -103,6 +103,12 @@
 
     run('if [ ! -d {{deploy_path}} ]; then mkdir -p {{deploy_path}}; fi');
 
+    // Check for existing /current directory (not symlink)
+    $result = run('if [ ! -L {{deploy_path}}/current ] && [ -d {{deploy_path}}/current ]; then echo true; fi')->toBool();
+    if ($result) {
+        throw new RuntimeException('There already is a directory (not symlink) named "current" in ' . env('deploy_path') . '. Remove this directory so it can be replaced with a symlink for atomic deployments.');
```
Comment (mbrodala) [inline recipe/common.php]: Did you mean `\RuntimeException` instead? ;-)

Comment (elfet) [inline recipe/common.php]: now ton in namespace. May be problem for v4.

Comment (vicdelfant) [inline recipe/common.php]: Fixed in 7ecee77. Maybe there should be another PR to fix the other occurrences of RuntimeException without an explicit namespace ;)
